bench/wtperf/wtperf.c

```diff
@@ -456,6 +486,79 @@ worker(void *arg)
 		goto err;
 	}
 
+	/* Setup for truncate */
+	if (thread->workload->truncate != 0) {
+
+		STAILQ_INIT(&truncate_stone_head);
+
+		cursor = cursors[cfg->icount % cfg->table_count];
+
+		/* Truncation percentage value. eg 10% is 0.1 */
+		truncation_percentage =
+		    (double)thread->workload->truncate_pct / 100;
+		/* How many entries between each milestone */
+		truncate_milestone_gap =
+		    thread->workload->truncate_count*truncation_percentage;
+		/* How many milestones we need */
+		needed_milestones = thread->workload->truncate_count /
+		    truncate_milestone_gap;
+
+		/* Reset this value for use again */
+		truncate_milestone_gap = 0;
+
+		cursor->next(cursor);
+		ret = cursor->get_key(cursor, &truncate_buf);
+
+		/* We have data */
+		if (ret == 0) {
```

Comment:
I'd switch the order of error checking here. Make it:

```
if (ret != 0) {
    lprintf(...)
    goto err;
}
```

Then you don't need to make the following code conditional on success.